### PR TITLE
Add a template tab for Oneoffixx templates

### LINKED ISSUE #5247: OneOffixx: Vorlagen Tab
Unter "Vorlagen" sollen auch die OneOffixx-Vorlagen in einem separaten Tab aufgelistet werden.

### PR DESCRIPTION
**Pull request by Rotonen**
Added a feature flagged view-only tab to list Oneoffixx templates.

I'm not yet making enabling Oneoffixx drop the normal document templates tab as it'd probably not yet be what is actually wanted.

Closes #5247

### CODE REVIEW

File: opengever/dossier/templatefolder/tabs.py [hunk 1]
```diff
@@ -197,3 +205,35 @@ class TemplateFolderDossierTemplates(BaseCatalogListingTab):
...
     major_actions = []
+
+
+def oneoffixx_css(item, value):
+    # Adapted from opengever/document/base.py
+    mtr = api.portal.get_tool('mimetypes_registry')
+    mimetypeitem = mimetype_lookup(mtr, item.get('content_type'))
+    icon = mimetypeitem[0].icon_path
+
+    # Adapted fromrom opengever/base/browser/helper.py
```
Comment (njohner) [inline opengever/dossier/templatefolder/tabs.py]: `# Adapted from opengever/base/browser/helper.py`

File: opengever/dossier/templatefolder/tabs.py [hunk 2]
```diff
@@ -197,3 +205,35 @@ class TemplateFolderDossierTemplates(BaseCatalogListingTab):
...
+    implements(IOneoffixxTableSourceConfig)
+
+    enabled_actions = []
+    major_actions = []
+    minor_actions = []
+
+    show_selects = False
+
+    columns = (
+        {'column': 'title', 'column_title': _(u'label_title', default=u'Title'), 'width': 300, 'transform': oneoffixx_css},
```
Comment (njohner) [inline opengever/dossier/templatefolder/tabs.py]: Maybe a linebreak here for readability

File: opengever/oneoffixx/tests/test_oneoffixx.py [hunk 3]
```diff
@@ -99,6 +117,33 @@ def test_template_id_stored_in_annotations(self, browser):
         annotations = IAnnotations(browser.context)
         self.assertEqual(node.get("value"), annotations['template-id'])
 
+    @browsing
+    def test_show_oneoffixx_templates_tab(self, browser):
+        self.login(self.regular_user, browser)
+        browser.open(self.templates)
+        expected_template_tabs = ['Oneoffixx templates', 'Local Templates', 'Tasktemplate Folders']
+        self.assertEqual(expected_template_tabs,  browser.css('.formTab').text)
```
Comment (njohner) [inline opengever/oneoffixx/tests/test_oneoffixx.py]: You could test the tab is not shown when the feature is disabled

File: plonetheme/teamraum/resources/css/gever/partials/_mimetypes.scss [hunk 4]
```diff
@@ -37,6 +37,18 @@ a[class*="icon-"] {
   }
 }
 
+// Spans have the text inside the element instead of next to it
+span[class*="icon-"] {
```
Comment (njohner) [inline plonetheme/teamraum/resources/css/gever/partials/_mimetypes.scss]: There is already styling for that selector above (https://github.com/4teamwork/opengever.core/blob/77868c7ce345b64cb3bcca8ec0765d4b04e2166e/plonetheme/teamraum/resources/css/gever/partials/_mimetypes.scss#L22-L26). You could merge these for better readability.

**Comment by Rotonen** [comment]
Added sortability and searchability. Ready for rereview.

**Comment by njohner** [comment]
LGTM

**Comment by Rotonen** [comment]
And now with better testing for the sorting and the filters.

**Comment by njohner** [comment]
nice 👍

File: opengever/dossier/locales/de/LC_MESSAGES/opengever.dossier.po [hunk 5]
```diff
@@ -628,6 +628,11 @@ msgstr "Label"
 msgid "label_linked_dossiers"
 msgstr "Verlinkte Dossiers"
 
+#. Default: "Local Templates"
+#: ./opengever/dossier/browser/tabbed.py
+msgid "label_local_templates"
+msgstr "Lokale Dokumentvorlagen"
```
Comment (deiferni) [inline opengever/dossier/locales/de/LC_MESSAGES/opengever.dossier.po]: i've just discussed this with the customer and also internally, we'd like to keep the labels a bit simpler, this here should just be "Dokumente" or "Documents".

File: opengever/dossier/locales/de/LC_MESSAGES/opengever.dossier.po [hunk 6]
```diff
@@ -664,6 +669,11 @@ msgstr "Das Dokument «${name}» befindet sich in einer Aufgabe und ist nicht ve
 msgid "label_number_of_containers"
 msgstr "Anzahl Behältnisse"
 
+#. Default: "Oneoffixx templates"
+#: ./opengever/dossier/browser/tabbed.py
+msgid "label_oneoffixx_templates"
+msgstr "Oneoffixx Vorlagen"
```
Comment (deiferni) [inline opengever/dossier/locales/de/LC_MESSAGES/opengever.dossier.po]: … and here just OneOffixx. mind the camel-casing, i think that is the correct way that product/company name is spelled, at least according to https://oneoffixx.com/.

**Comment by Rotonen** [comment]
Ready for rereview.

File: opengever/dossier/locales/de/LC_MESSAGES/opengever.dossier.po [hunk 7]
```diff
@@ -664,6 +669,11 @@ msgstr "Das Dokument «${name}» befindet sich in einer Aufgabe und ist nicht ve
 msgid "label_number_of_containers"
 msgstr "Anzahl Behältnisse"
 
+#. Default: "OneOffixx templates"
+#: ./opengever/dossier/browser/tabbed.py
+msgid "label_oneoffixx_templates"
+msgstr "OneOffixx Vorlagen"
```
Comment (deiferni) [inline opengever/dossier/locales/de/LC_MESSAGES/opengever.dossier.po]: uh, the " Vorlagen" is still in there, the label should be just "OneOffixx".

**Comment by Rotonen** [comment]
Ready for rereview.

**Pull request merged**